### PR TITLE
File page index fix

### LINKED ISSUE #1677: getting urls to specific pages
@waxlamp - a related question to https://github.com/dandi/dandi-cli/issues/1325 - how do i get a url to the second page?

for example, in dandiset 26, it would be really hard to navigate to a specific page given the number of assets there within a specific tree level for spim.

### PR DESCRIPTION
**Pull request by marySalvi**
Closes #1677 
This PR adds the page index value to the route query.
Please also note that I removed a few watch statements in favor of reactive values as the watch statements were causing multiple calls with the added query value. I think its generally good practice to avoid watches when possible.

### CODE REVIEW

**Review by mvandenburgh — COMMENTED** [review]
@marySalvi it seems this doesn't work if you try to navigate directly to a page, i.e. clicking on https://deploy-preview-1704--gui-staging-dandiarchive-org.netlify.app/dandiset/209782/draft/files?location=sub-CRS2&page=3 brings me to page 1, even though the URL specifies page 3.

**Comment by marySalvi** [comment]
> @marySalvi it seems this doesn't work if you try to navigate directly to a page, i.e. clicking on https://deploy-preview-1704--gui-staging-dandiarchive-org.netlify.app/dandiset/209782/draft/files?location=sub-CRS2&page=3 brings me to page 1, even though the URL specifies page 3.

Good catch. It is actually the 3rd page of results its just that pagination text field isn't updated correctly. I'll fix that.

**Comment by marySalvi** [comment]
> > @marySalvi it seems this doesn't work if you try to navigate directly to a page, i.e. clicking on https://deploy-preview-1704--gui-staging-dandiarchive-org.netlify.app/dandiset/209782/draft/files?location=sub-CRS2&page=3 brings me to page 1, even though the URL specifies page 3.
> 
> Good catch. It is actually the 3rd page of results its just that pagination text field isn't updated correctly. I'll fix that.

fixed in 6e781533347d93266375292b4def17674a7f1236

File: web/src/views/FileBrowserView/FileBrowser.vue [hunk 1]
```diff
@@ -517,9 +518,13 @@ watch(() => route.query, (newRouteQuery) => {
...
-
-// fetch new page of items when a new one is selected
-watch(page, getItems);
+const changePage = (newPage: number) => {
+  page.value = newPage;
+  router.push({
+    ...route,
+    query: { location: location.value, page: String(page.value) },
+  } as RawLocation);
+};
```
[the remark this answers is not in the document]
Comment (marySalvi) [reply web/src/views/FileBrowserView/FileBrowser.vue]: addressed in 6a6e4bd9384c40ab89938a3e099b1f013dcdfd9d

**Review by mvandenburgh — APPROVED** [review]
LGTM!

**Pull request merged**